### ISSUE TITLE
[Event queue] Failures due to error fetching the shop's config.json

### ISSUE THREAD

**Issue by franckc** [OriginProtocol/dshop#373]
There are 2 jobs in the events queue in prod that are failing due to not being able to load the associated shop's config. 

Here is the error:
```
request to https://sweet-tea-tumblers.ogn.app/sweet-tea-tumblers/config.json failed, reason: getaddrinfo ENOTFOUND sweet-tea-tumblers.ogn.app sweet-tea-tumblers.ogn.app:443
```

The URL https://sweet-tea-tumblers.ogn.app fails resolving.

**Comment by franckc** [comment]
This is shop id 211

The queue failed for those 2 jobs:
 - job id 0x761dac419b570bbf0157b05b8ab629e6651703c1e363af1a8b17013141e3ba23, which was mined on Aug-04-2020 09:30 PM +UTC
 - job id 0x716c45d2a0bd7814ca9e5b489aebe58e3ff07bee98135a545ec14bf35218a201, which was mined on Aug-04-2020 10:02 PM +UTC

The deployment for this shop failed a few times due to #352 

As a result, based on the data in the shop_deployments table, the shop must have been down during the following time windows:
 - from 2020-08-01 10:12pm UTC to 2020-08-02 3:59pm UTC
 - from 2020-08-02 4:03pm UTC to 2020-08-04 01:17am UTC

It's also interesting to note that the shop's domain was changed from https://sweet-pea-tumblers.ogn.app to https://sweet-pea-shop.ogn.app on 2020-08-01 01:43am UTC.

For some reason, the event processor is still trying to process the events using the old domain https://sweet-pea-tumblers.ogn.app. E.g. it's trying to fetch https://sweet-tea-tumblers.ogn.app/sweet-tea-tumblers/config.json (which won't work) as opposed to using the correct URL: https://sweet-pea-shop.ogn.app/sweet-tea-tumblers/config.json

**Comment by franckc** [comment]
The shop's config dataUrl field is still pointing to the old domain:
```
node configCli.js --networkId=1 --shopId=211 --operation=get --key=dataUrl
dshop.cli: Loaded shop Sweet Pea Shop (211)
dshop.cli: Shop 211 Sweet Pea Shop config: dataUrl=https://sweet-tea-tumblers.ogn.app/sweet-tea-tumblers/
dshop.cli: Finished
```

Which causes the event processor to fail.

I'm not sure why that config did not get updated?

**Comment by nick** [comment]
I'm thinking we should make sure config.json is correct at the time of deploy by making sure backendUrl, listingId etc are all up to date.

**Comment by franckc** [comment]
Agreed for more data consistency checks as part of the deployment.

Now, just to be clear, in this case it is the shop's config in the database that is incorrect... not the config.json

**Comment by mikeshultz** [comment]
Guessing DNS was updated without a deploy?  Is config.json something we'd want to pull from an IPFS hash rather than DNS?  Just asking outloud.

**Comment by franckc** [comment]
I manually fixed the dataUrl for shop Id 211.
I also re-reprocessed successfully the 2 orders that were failing.

After looking at the code, it looks like we are not updating dataUrl in the config in case the merchant updates their shop's hostname. I'll send a PR to fix that...

**Comment by franckc** [comment]
I manually fixed the dataUrl for shop Id 211.
I also re-reprocessed successfully the 2 orders that were failing.

After looking at the code, it looks like we are not updating dataUrl in the config in case the merchant updates their shop's hostname. I'll send a PR to fix that...

**Comment by franckc** [comment]
@gullible1 FYI Orders on the "Sweet Pea Shop" should now fixed. You can let the merchant know. It broke because the merchant changed the shop's hostname (under Settings > Appearance > Store Domain) which is totally fine but that triggered a bug on our side that I will work on fixing...

**Comment by gullible1** [comment]
You legend, @franckc ! He will be happy to hear that - I'll ping him now.

**Comment by franckc** [comment]
Fixed in #402
